verify/cert.go

```diff
@@ -103,10 +125,35 @@ func (cv *CertVerifier) VerifyMerkleProof(inclusionProof []byte, root []byte,
 }
 
 // fetches a block number provided a subtraction of a user defined conf depth from latest block
-func (cv *CertVerifier) getConfDeepBlockNumber() (*big.Int, error) {
-	blockNumber, err := cv.ethClient.BlockNumber(context.Background())
+func (cv *CertVerifier) getConfDeepBlockNumber(ctx context.Context) (*big.Int, error) {
+	if cv.waitForFinalization {
+		// https://github.com/ethereum/execution-apis/blob/4140e528360fea53c34a766d86a000c6c039100e/src/eth/block.yaml#L61
+		var header = types.Header{}
+		err := cv.ethClient.Client().CallContext(ctx, &header, "eth_getBlockByNumber", "finalized", false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get finalized block: %w", err)
+		}
+		return header.Number, nil
+	}
+	blockNumber, err := cv.ethClient.BlockNumber(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get latest block number: %w", err)
 	}
-	return new(big.Int).SetUint64(max(blockNumber-cv.ethConfirmationDepth, 0)), nil
+	if blockNumber < cv.ethConfirmationDepth {
+		return big.NewInt(0), nil
+	}
+	return new(big.Int).SetUint64(blockNumber - cv.ethConfirmationDepth), nil
+}
+
+// retrieveBatchMetadataHash retrieves the batch metadata hash stored on-chain at a specific blockNumber for a given batchID
+// returns an error if some problem calling the contract happens, or the hash is not found
+func (cv *CertVerifier) retrieveBatchMetadataHash(ctx context.Context, batchID uint32, blockNumber *big.Int) ([32]byte, error) {
+	onchainHash, err := cv.manager.BatchIdToBatchMetadataHash(&bind.CallOpts{Context: ctx, BlockNumber: blockNumber}, batchID)
+	if err != nil {
+		return [32]byte{}, fmt.Errorf("calling EigenDAServiceManager.BatchIdToBatchMetadataHash: %w", err)
```

Comment:
knit
```suggestion
		return [32]byte{}, fmt.Errorf("error calling EigenDAServiceManager.BatchIdToBatchMetadataHash: %w", err)
```

Reply:
Disagree here. Read https://github.com/uber-go/guide/blob/master/style.md#error-wrapping last week which makes a lot of sense, been trying to use this convention as much as possible.